utils/debug/api.go

```diff
@@ -26,14 +29,49 @@ func GetExecutionAPIBlockHeader(
 		return nil, err
 	}
 
-	return &flow.Header{
-		HeaderBody: flow.HeaderBody{
-			ChainID:   flow.ChainID(resp.Block.ChainId),
-			ParentID:  flow.Identifier(resp.Block.ParentId),
-			Height:    resp.Block.Height,
-			Timestamp: resp.Block.Timestamp.AsTime(),
-		},
-	}, nil
+	blockHeader, err := rpcConvert.MessageToBlockHeader(resp.Block)
+	if err != nil {
+		return nil, err
+	}
+	untrusted := flow.UntrustedHeaderBody{
+		ChainID:            blockHeader.ChainID,
+		ParentID:           blockHeader.ParentID,
+		Height:             blockHeader.Height,
+		Timestamp:          blockHeader.Timestamp,
+		ParentVoterIndices: blockHeader.ParentVoterIndices,
+		ParentVoterSigData: blockHeader.ParentVoterSigData,
+		ProposerID:         blockHeader.ProposerID,
+		LastViewTC:         blockHeader.LastViewTC,
+	}
+	if !blockHeader.ContainsParentQC() {
```

Comment:
I don't understand this part. `MessageToBlockHeader` already returns a converted value which is pre-validated by using constructors. We we are essentially doubling the work and doing the same here?